From: Friedemann Kleint <Friedemann.Kleint@qt.io>
Date: Tue, 12 Oct 2021 09:33:53 +0200
Subject: shiboken6: Adapt to LLVM 12.01

The identifier for anonymous enums has been changed, causing
shiboken to no longer recognize them.

(cherry picked from commit 1948a1b6f9a3547ba9fba9b8c525eb6e1c6b4f13)
---
 sources/shiboken2/ApiExtractor/clangparser/clangbuilder.cpp | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/sources/shiboken2/ApiExtractor/clangparser/clangbuilder.cpp b/sources/shiboken2/ApiExtractor/clangparser/clangbuilder.cpp
index 73b1aca..c6f3bb8 100644
--- a/sources/shiboken2/ApiExtractor/clangparser/clangbuilder.cpp
+++ b/sources/shiboken2/ApiExtractor/clangparser/clangbuilder.cpp
@@ -887,8 +887,10 @@ static QString enumType(const CXCursor &cursor)
         // PYSIDE-1228: For "typedef enum { v1, v2 } Foo;", type will return
         // "Foo" as expected. Care must be taken to exclude real anonymous enums.
         name = getTypeName(clang_getCursorType(cursor));
-        if (name.contains(QLatin1String("(anonymous")))
+        if (name.contains(u"(unnamed") // Clang 12.0.1
+            || name.contains(u"(anonymous")) { // earlier
             name.clear();
+        }
     }
     return name;
 }
